include/seqan3/io/sequence_file/format_embl.hpp

```diff
@@ -0,0 +1,347 @@
+               seq_type                       && sequence,
+               id_type                        && id,
+               qual_type                      && SEQAN3_DOXYGEN_ONLY(qualities))
+    {
+
+        ranges::ostreambuf_iterator stream_it{stream};
+        size_t sequence_size = 0;
+        if constexpr (!detail::decays_to_ignore_v<seq_type>)
+            sequence_size = ranges::size(sequence);
+
+        // ID
+        if constexpr (detail::decays_to_ignore_v<id_type>)
+        {
+            throw std::logic_error{"The ID field may not be set to ignore when writing embl files."};
+        }
+        else
+        {
+            if (options.complete_header)
+            {
+                ranges::copy(id, stream_it);
+            }
+            else
+            {
+                if (ranges::empty(id)) //[[unlikely]]
+                    throw std::runtime_error{"The ID field may not be empty when writing embl files."};
+
+                stream_it = 'I';
+                stream_it = 'D';
+                stream_it = ' ';
+                ranges::copy(id, stream_it);
+                stream_it = ';';
+                stream_it = ' ';
+                ranges::copy(std::to_string(sequence_size), stream_it);
+                ranges::copy(std::string_view{" BP.\nXX\n"}, stream_it);
+            }
+
+        }
+
+        // Sequence
+        if constexpr (detail::decays_to_ignore_v<seq_type>) // sequence
+        {
+            throw std::logic_error{"The SEQ and SEQ_QUAL fields may not both be set to ignore when writing embl files."};
+        }
+        else
+        {
+            if (ranges::empty(sequence)) //[[unlikely]]
+                throw std::runtime_error{"The SEQ field may not be empty when writing embl files."};
+
+            ranges::copy(std::string_view{"SQ Sequence "}, stream_it);
+            ranges::copy(std::to_string(sequence_size), stream_it);
+            ranges::copy(std::string_view{" BP;"}, stream_it);
+            stream_it = '\n';
+            auto seq = sequence | ranges::view::chunk(60);
+            unsigned int i = 0;
+            size_t bp = 0;
+            while (bp < sequence_size)
+            {
+                ranges::copy(seq[i] | view::to_char
+                                    | ranges::view::chunk(10)
+                                    | ranges::view::join(' '), stream_it);
+                bp = std::min(sequence_size, bp + 60);
+                ++i;
+                stream_it = ' ';
+                if (bp % 60 > 0)
+                {
+                    for(int j = bp; j < (60 * i); j++)
+                    {
+                        stream_it = ' ';
+                        if (j % 10 == 0)
+                            stream_it = ' ';
+                    }
+                    ranges::copy(std::to_string(bp), stream_it);
+
+                }
+                else
+                {
+                    ranges::copy(std::to_string(bp), stream_it);
```

Comment:
duplicate in both if/else branches: move below

Reply:
Actually, you can omit the whole condition and loop, because you can calculate the number of blanks with the following expression:
```cpp
uint8_t num_blanks = 60 * i - bp;  // for sequence characters
num_blanks += num_blanks / 10;     // additional chunk separators
```
For full line length (bp = 60*i) it will be 0, so no case distinction is needed.